sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusSenderAsyncClient.java

```diff
@@ -426,6 +486,46 @@ public void close() {
         });
     }
 
+
+    private Flux<Long> scheduleMessagesInternal(Iterable<ServiceBusMessage> messages,
+        OffsetDateTime scheduledEnqueueTime, ServiceBusTransactionContext transaction) {
+        if (Objects.isNull(messages)) {
+            return fluxError(logger, new NullPointerException("'messages' cannot be null."));
+        }
+
+        if (Objects.isNull(scheduledEnqueueTime)) {
+            return fluxError(logger, new NullPointerException("'scheduledEnqueueTime' cannot be null."));
+        }
+
+        return createBatch().flatMapMany(messageBatch -> {
+            messages.forEach(message -> messageBatch.tryAdd(message));
+            return scheduleMessagesInternal(messageBatch, scheduledEnqueueTime, transaction);
+        });
+    }
+
+    private Flux<Long> scheduleMessagesInternal(ServiceBusMessageBatch message, OffsetDateTime scheduledEnqueueTime,
+        ServiceBusTransactionContext transactionContext) {
+        if (Objects.isNull(message)) {
+            return fluxError(logger, new NullPointerException("'message' cannot be null."));
+        }
+
+        if (Objects.isNull(scheduledEnqueueTime)) {
+            return fluxError(logger, new NullPointerException("'scheduledEnqueueTime' cannot be null."));
+        }
+
+        return getSendLink()
+            .flatMapMany(link -> link.getLinkSize().flatMapMany(size -> {
+                int maxSize =  size > 0
```

Comment:
Why are we getting link size again instead of just using the batch's max size???